### DISCUSSION TITLE
Add support for "alias" field in UI mode

### DISCUSSION THREAD

**Discussion by Mincka** [home-assistant/frontend#8622] [Automations / scripts / scenes]
This topic is a follow up for what it looks like a WIP regarding the implementation of "alias" in UI mode.

My heart hesitates between keeping the automations in folders in YAML or fully use the UI. With the recent improvements, more and more can be done with the UI et it will be also easier to organize automations and scripts with area mapping. Adding "alias" in the GUI will allow users to add comments in complex automations. The only downside I see is that it could often be an unnecessary field and showing it by default would add complexity for end-users.

Following the [recent support for "alias" for steps in automation and scripts](https://github.com/home-assistant/core/pull/46647), the next step would be to add support for "alias" in UI mode since [it already exists in YAML mode](https://github.com/home-assistant/frontend/pull/7153).

I saw [here](https://github.com/home-assistant/frontend/pull/7153#issuecomment-702875145) that @donkawechico wanted to implement it but I am not sure of the current status, that's why I am opening this thread.

**Comment by donkawechico** [comment]
Just saw this. Yeah, I did the initial work for expanding alias out to scripts and automations in core in order to resolve my own [WTH?](https://community.home-assistant.io/t/wth-cant-we-annotate-sequence-steps-for-logging-executing-step-call-service-make-lights-blue/224447) post.

I know I started a branch or a PR somewhere that added the alias to the UI in some fashion, but I put that project aside to do other stuff, and now I can't find it :P 

I agree that adding an alias field could clutter up the UI if no one's using it, and I designed around that by using a collapsible card, and adding some kind of "Edit Alias" option to the three-dot menu.

I still believe this would be a valuable addition to the UI -- not so much for debugging automations anymore, now that we have the new automation tracing stuff -- but for making complex automations more readable and scannable by human eyes.

I'm sure I'll get back to it eventually.

**Reply by PGutti** [reply]
There is already a "hidden" field in the UI editor: The Trigger ID!  This field was moved to the `...` menu (upper right) in one of the recent updates..  A additional comment field could be handled the same. Yes, when it is used by a programmer it uses space on the screen. But this is up to the programmer.  The field could be a container like the one i'm currently writing into: Variable size field.  That ends up in a YAML statement `###`.

BTW: The name "Alias" is confusing. I would name this field "Comment".
